docker/Dockerfile

```diff
@@ -214,7 +214,14 @@ RUN source /opt/ros/$ROS_DISTRO/setup.bash && \
 ARG RMW_IMPLEMENTATION="rmw_cyclonedds_cpp"
 ENV RMW_IMPLEMENTATION=${RMW_IMPLEMENTATION}
 RUN source /opt/ros/$ROS_DISTRO/setup.bash && \
-    if [[ "$ROS_VERSION" == "2" ]]; then \
+    if [[ "$RMW_IMPLEMENTATION" == "rmw_zenoh" ]]; then \
+        git clone https://github.com/ros2/rmw_zenoh.git $WORKSPACE/src/upstream/rmw_zenoh && \
+        rosdep init || true && \
+        rosdep update --rosdistro $ROS_DISTRO && \
+        apt-get update && \
+        rosdep install -y --from-paths $WORKSPACE/src/upstream/rmw_zenoh --ignore-src --rosdistro $ROS_DISTRO && \
+        rm -rf /var/lib/apt/lists/* ; \
```

Comment:
We could save some lines of code by just cloning the repo as upstream repo in the `dependencies` stage (according to the vcs-repos), but would be less explicit. What do you think?

Reply:
Not sure if that could give us less lines, because you would also need to check for an existing .repos, add some lines there, ... would keep it here with the other middlewares.